sdk/nodejs/errors.ts

```diff
@@ -81,3 +81,53 @@ export function isGrpcError(err: Error): boolean {
     const code = (<any>err).code;
     return code === grpc.status.UNAVAILABLE || code === grpc.status.CANCELLED;
 }
+
+interface InputPropertyErrorArgs {
+    message: string;
+    propertyPath: string;
+    reason: string;
```

Comment:
The error message would look like:

```
error: pulumi:providers:some-type resource 'some-name' has a problem: failed initializing:\n" 
			- property testproperty with value '{testvalue}' has a problem: property not valid because foo
```

With the message being the overall failure reason, and the reason being more specific.  We could change the message the engine emits here though.

Reply:
Maybe just having it without the message would be fine on second thought.  That would result in an error message like:

```
error: pulumi:providers:some-type resource 'some-name' has a problem:" 
		- property testproperty with value '{testvalue}' has a problem: property not valid because foo
```

which is probably enough.

Reply:
Changed it to the latter version now.